src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -841,10 +841,36 @@ public interface IConversionExpression : IHasOperatorMethodExpression
         /// Kind of conversion.
         /// </summary>
         ConversionKind ConversionKind { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an identity conversion.
+        /// </summary>
+        bool IsIdentity { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit ('widening' in VB) conversion.
+        /// </summary>
+        bool IsImplicit { get; }
```

Comment:
The terms "implicit" and "explicit" are confusing because they have meaning in one of our source languages with respect to possible conversions, and another meaning with respect to actual conversions.

"PresentInSource" or "HasCastInSource" strike me as OK for indicating that a conversion is explicitly indicated in source, and "IsNarrowing" and "IsWidening" string me as OK for indicating that a conversion is narrowing or widening.

IsNarrowing does not imply PresentInSource for VB.

One could argue that whether or not a cast is present in source is not semantically significant.